src/uploader.c

```diff
@@ -327,6 +343,15 @@ static void create_bucket_entry(uv_work_t *work)
 
     json_object_object_add(body, "hmac", hmac);
 
+    // TODO: Is it okay for type to be NULL?
```

Comment:
It's okay for erasure to be `undefined`, however `null` isn't valid erasure type, which I think may be causing the issue, and would be fixed by changing these lines to:
```js
if (state->rs) {
  struct json_object *erasure = json_object_new_object(); 
  json_object *erasure_type = json_object_new_string("reedsolomon");
  json_object_object_add(erasure, "type", erasure_type);
  json_object_object_add(body, "erasure", erasure);
}
```